shared/actions/people.js

```diff
@@ -0,0 +1,185 @@
+// @flow
+import * as PeopleGen from './people-gen'
+import * as Saga from '../util/saga'
+import * as I from 'immutable'
+import * as Constants from '../constants/people'
+import * as Types from '../constants/types/people'
+import * as RouteTypes from '../constants/types/route-tree'
+import * as RouteConstants from '../constants/route-tree'
+import * as RPCTypes from '../constants/types/rpc-gen'
+import engine from '../engine'
+import {peopleTab} from '../constants/tabs'
+import {type TypedState} from '../constants/reducer'
+import {createDecrementWaiting, createIncrementWaiting} from '../actions/waiting-gen'
+
+const _getPeopleData = function(action: PeopleGen.GetPeopleDataPayload, state: TypedState) {
+  return Saga.all([
+    Saga.call(RPCTypes.homeHomeGetScreenRpcPromise, {
+      markViewed: action.payload.markViewed,
+      numFollowSuggestionsWanted: action.payload.numFollowSuggestionsWanted,
+    }),
+    Saga.identity(state.config.following),
+    Saga.identity(state.config.followers),
+    Saga.put(createIncrementWaiting({key: Constants.getPeopleDataWaitingKey})),
```

Comment:
This is getting stuck somewhat often, resulting in a perma-spinner. A solution would be to make `_processPeopleData` a generator and use a try/catch, but I'd like to avoid that. So far I have a `decrementWaiting` at the end of `_processPeopleData` (success case) and passed to `safeTakeLatestPure` (error case) :(

Reply:
i don't think you want all here, i think sequentially is likely what you want and incrementwaiting first